libs/sdk-ui-dashboard/src/presentation/filterBar/attributeFilter/dashboardDropdownBody/configuration/limitValues/shared/limitingItemsHook.ts

```diff
@@ -308,18 +337,59 @@ const getDatasetTitle = (
     dateDataSet: ObjRef | undefined,
     defaultDateFilterTitle: string,
     filterConfigByDimension: IDashboardDateFilterConfigItem[],
-    intl: IntlShape,
     filterConfig?: IDashboardDateFilterConfig,
 ) => {
     const usedConfig = dateDataSet
         ? filterConfigByDimension.find((config) => areObjRefsEqual(config.dateDataSet, dateDataSet))?.config
         : filterConfig;
 
-    const dataSetTitle =
-        !usedConfig || usedConfig?.filterName === "" ? defaultDateFilterTitle : usedConfig?.filterName;
+    return !usedConfig || usedConfig?.filterName === "" ? defaultDateFilterTitle : usedConfig?.filterName;
+};
+
+export const useCommonDateItems = (
+    availableDatasets: ICatalogDateDataset[],
+    dependentCommonDateFilter: IDashboardDateFilter,
+    dependentDateFilters?: IDashboardDependentDateFilter[],
+): IValuesLimitingItemWithTitle[] => {
+    return useMemo(() => {
+        return availableDatasets
+            .filter((availableDataset) => {
+                if (dependentDateFilters) {
```

Comment:
You can just do `.filter((availableDataset) => !dependentDateFilters.some(...))` and save lots of lines and have it more streamlined and readable. No reason for explicit `return [boolean]` which is always bit suspicious IMHO.

Reply:
Replaced it.